### DISCUSSION TITLE
Signing in user from server-side

### DISCUSSION THREAD

**Discussion by ryan-mccaffrey** [nextauthjs/next-auth#10140] [Help]
So i know that the `signIn` function provided by next-auth is only meant to be used to sign in a user on the client side, but under the hood it looks like it's making a POST to `${baseUrl}/callback/credentials` for a credentials sign in, which then establishes a new user session (and seems to be setting the session-token in cookies for future requests).

I've tried creating an internal `signIn` helper that can be used server-side by not relying on the `window` var, and this helper also makes a POST request to `${baseUrl}/callback/credentials` to sign in the user. When the fetch is made, I get a 200 response with the response payload: `{ url: 'http://localhost:3000/api/auth/signin?csrf=true' }` but no session-token cookie is set, and calls to `getSession()` and `getServerSession()` both result in null values.

An example of the post looks like:
```
const res = await fetch("http://localhost:3000/api/auth/callback/credentials", {
    method: "post",
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
    },
    // @ts-expect-error
    body: new URLSearchParams({
      email: "email.address@gmail.com",
      password: "password",
      redirect: false,
      csrfToken: await getCsrfToken(),
      callbackUrl: "http://localhost:3000",
      json: true,
    })
  });
```

The only cookies that are set as a result are:
```
[
    'next-auth.csrf-token',
    'dcf98d148cd3f28061135e8b69579619f8ee1efcbeedbda1c074d05618864af9|77eee29842f6f75e63b97c5f81e5b50d3ba7d56409e0c610b3e55e77422dc867'
],
[ 'next-auth.callback-url', 'http://localhost:3000/' ]
```

Additionally, when I put logs into the definition of the `authorize` function for the credentials provider, I can see that it _is_ logging when using the next-auth `signIn` helper, but not when calling the fetch manually, making it seem like this request isn't being handled the same way

**Comment by llopez** [comment]
Hey. I am trying to do the same here.

I would like to be able to have my own signin form and use server actions because I guess we dont need client logic for this page.

I could not find the way. The docs dont say to much about it.

**Comment by livingmine** [comment]
@ryan-mccaffrey @llopez You can always get server side `signIn` method by destructuring the response of the `NextAuth` initialization like the following.
```typescript
const {
  handlers: { GET, POST },
  auth,
  signIn, // your server side signIn method
} = NextAuth((request) => {...}
```
